### PR TITLE
Add test scneario for settlement chain

### PR DESCRIPTION
**Pull request by junha1**

### CODE REVIEW

File: settlement/src/lib.rs [hunk 1]
```diff
@@ -30,24 +31,30 @@ pub trait SettlementChain: Send + Sync {
...
+    ) -> Result<Decimal, Error>;
 
     /// Returns the current balance of a particular non-fungible token collection in the treasury contract,
     /// identified as their token indices.
     async fn get_treasury_non_fungible_token_balance(
         &self,
-        address: String,
-    ) -> Result<Vec<String>, Error>;
+        address: HexSerializedVec,
+    ) -> Result<Vec<HexSerializedVec>, Error>;
```
Comment (hyeonLewis) [inline settlement/src/lib.rs:57]: Does it return `Vec<HexSerizlizedVec>`? (string to HexSerializedVec for token indices?)

File: settlement/src/lib.rs [hunk 2]
```diff
@@ -68,4 +75,25 @@ pub trait SettlementChain: Send + Sync {
...
+        address: HexSerializedVec,
+        token_address: HexSerializedVec,
+    ) -> Result<Decimal, Error>;
+
+    /// Submits a transaction to transfer a fungible token
+    /// from the given externally owned account to the given receiver address.
+    async fn eoa_transfer_fungible_token(
+        &self,
+        address: HexSerializedVec,
+        private_key: HexSerializedVec,
```
Comment (hyeonLewis) [inline settlement/src/lib.rs:94]: `private_key` -> `sender_private_key` looks more clear.

File: settlement/src/tests.rs [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    let contract_sequence = sc.get_contract_sequence().await.unwrap();
+    let initial_temporary_receiver_balance = sc
+        .eoa_get_fungible_token_balance(temporary_receiver_address.clone(), token_address.clone())
+        .await
+        .unwrap();
+    assert_eq!(initial_treasury_header, chain_info.last_finalized_header);
+    assert_eq!(contract_sequence, initial_contract_sequence);
+
+    // Apply transactions
+    let mut txes = Vec::new();
```
Comment (hyeonLewis) [inline settlement/src/tests.rs:70]: `txes` -> `txns` would be better.

Comment (junha1) [reply settlement/src/tests.rs:70]: Isn't `txes` more common? I saw it multiple times from multiple sources.

Comment (hyeonLewis) [reply settlement/src/tests.rs:70]: I think `txns` is more appropriate. For example, google gives me about `taxes` when using `txes`, and `transactions` for `txns`.

Comment (junha1) [reply settlement/src/tests.rs:70]: Ok I think you're right. However I just realized that `transactions` seems the best 😁

File: settlement/src/lib.rs (hunk 1 above)
Comment (junha1) [reply settlement/src/lib.rs:57]: Yes. As I mentioned before, NFT index could be in Hash (not an integer).
This method won't be implemented soon anyway, so let's just skip for now.

File: settlement/src/lib.rs (hunk 2 above)
Comment (junha1) [reply settlement/src/lib.rs:94]: fixed

File: settlement/src/tests.rs (hunk 3 above)
Comment (junha1) [reply settlement/src/tests.rs:70]: fixed

File: settlement/src/tests.rs [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+) {
+    // Setup the on-chain state
+    let mut csv = CommitSequenceVerifier::new(
+        chain_info.last_finalized_header.clone(),
+        chain_info.reserved_state.clone(),
+    )
+    .unwrap();
+
+    // Query the initial status
+    let intial_balance = sc
```
Comment (chemonoworld) [inline settlement/src/tests.rs:56]: intial -> initial

File: settlement/src/tests.rs [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+        csv.apply_commit(&Commit::Transaction(tx.clone())).unwrap();
+        transactions.push(tx);
+    }
+    let execute_tx = execution::create_execution_transaction(
+        &Execution {
+            target_chain: chain_info.chain_name,
+            contract_sequence: initial_contract_sequence,
+            message: ExecutionMessage::TransferFungibleToken(TransferFungibleToken {
+                token_address: token_address.clone(),
+                amount: intial_balance,
```
Comment (chemonoworld) [inline settlement/src/tests.rs:88]: intial -> initial

File: settlement/src/tests.rs [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+    // Check the result
+    assert_eq!(
+        sc.eoa_get_fungible_token_balance(
+            temporary_receiver_address.clone(),
+            token_address.clone()
+        )
+        .await
+        .unwrap(),
+        initial_temporary_receiver_balance + intial_balance
```
Comment (chemonoworld) [inline settlement/src/tests.rs:179]: intial -> initial

File: settlement/src/tests.rs [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+    let merkle_tree = OneshotMerkleTree::create(
+        commits[1..=(commits.len() - 2)]
+            .iter()
+            .map(|c| c.to_hash256())
+            .collect(),
+    );
+    let merkle_proof = merkle_tree
+        .create_merkle_proof(execute_tx.to_hash256())
+        .unwrap();
+    sc.execute(execute_tx, 1, merkle_proof).await.unwrap();
```
Comment (chemonoworld) [inline settlement/src/tests.rs:169]: The test probably fails if ```execute```or ```get_light_client_header``` is executed first while the tx from ```update_treasury_light_client``` is pending, doesn't it?

File: settlement/src/tests.rs (hunk 4 above)
Comment (junha1) [reply settlement/src/tests.rs:56]: fixed

File: settlement/src/tests.rs (hunk 5 above)
Comment (junha1) [reply settlement/src/tests.rs:88]: fixed

File: settlement/src/tests.rs (hunk 6 above)
Comment (junha1) [reply settlement/src/tests.rs:179]: fixed

File: settlement/src/tests.rs (hunk 7 above)
Comment (junha1) [reply settlement/src/tests.rs:169]: You're right. I added sleep between them.

**Comment by chemonoworld** [comment]
LGTM

**Review by chemonoworld — APPROVED** [review]

**Pull request merged**